#! /usr/bin/env python
PACKAGE="tuw_gamepad"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
                                                                        
gen.add("debug"    ,   bool_t, 0, "Prints debug messages", False )
gen.add("joy_msg_timeout",    int_t,    0, "[ms] Timeout for receiving joy messages, -1 no timeout",  1,  -1, 10)
gen.add("rate",    double_t,    0, "[Hz] command publish rate",  10,  0, 50)

gen.add("max_vx", double_t, 0, "[m/s] max translation velocity on X axis",    1., 0,   5)
gen.add("max_vy", double_t, 0, "[m/s] max translation velocity on Y axis",    0., 0,   5)
gen.add("max_vw", double_t, 0, "[rad/s] max rotation velocity on the around Z axis",    1., 0,   5)

gen.add("deadman_button",    int_t,    0, "discrete deadman button needed to enable the system", 5,  -1, 10)
gen.add("scale_button",    int_t,    0, "discrete to speed up the velocity by the factor scale", 4,  -1, 10)
gen.add("scale", double_t, 0, "speed up if the scale button is pressed",    0., 2,   5)
gen.add("passthrough_button",    int_t,    0, "passthrough button needed if you like to pass commands through your gamepad", 0,  -1, 10)

gen.add("axis_vx",    int_t,    0, "Button for analog translation velocity along X",  1,  -1, 10)
gen.add("axis_vy",    int_t,    0, "Button for analog translation velocity along Y", -1,  -1, 10)
gen.add("axis_vw",    int_t,    0, "Button for analog rotation velocity around Z", 0,  -1, 10)

gen.add("axis_vx_discrete",    int_t,    0, "Button for discrete translation velocity along X",  7,  -1, 10)
gen.add("axis_vy_discrete",    int_t,    0, "Button for discrete translation velocity along Y", -1, -1, 10)
gen.add("axis_vw_discrete",    int_t,    0, "Button for discrete rotation velocity around Z", 6,  -1, 10)

gen.add("wheel_radius", double_t, 0, "needed for IWS control: wheel radius",    0.075, 0,   1)
gen.add("wheel_displacement", double_t, 0, "needed for IWS control: distance between wheels",    0.449, 0,   2)

exit(gen.generate(PACKAGE, "tuw_gamepad", "GamepadControl"))
